src/helpers.c

```diff
@@ -1015,3 +1015,31 @@ void print_regexp_error(int errcode, regex_t *regexp)
 
 	    free(error_buffer);
 }
+
+bool is_url_allowed(char *url)
+{
+	bool insecure = false;
+
+	if (url) {
```

Comment:
I know that this is used in other parts of the code, but using a condition == NULL to select a different behavior of a function is confusing. Instead you could create a is_url_allowed() that returned true or false if the url is valid or not. and use that with content_url and verion_url when approprite.